### PR TITLE
Add mandatory id to rules & Update tests accordingly

### LINKED ISSUE #49: Add id field and correctly select rules in yaml
Add id field and correctly allow multiple rules to be defined in yaml.

### PR DESCRIPTION
**Pull request by The-Huginn**
Resolves #49 

In TriagePullRequestProcessor.java
@Valid annotation was not working in combination with @ConfigFile in method parameters. Maybe due to bug with nested Class. Hence a new method to invoke not null check with @Valid annotated parameter.

### CODE REVIEW

File: src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java [hunk 1]
```diff
@@ -42,5 +48,13 @@ void onPullRequestOpened(@PullRequest.Opened GHEventPayload.PullRequest pullRequ
         }
 
     }
+
+    boolean matchRule(GHPullRequest pullRequest, @Valid WildFlyRule rule) {
+        if (Matcher.matches(pullRequest, rule)) {
+            LOG.debug("Matched rule with id: " + rule.id + ".");
```
Comment (xstefank) [inline src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java:54]: do you know about debugf?

**Comment by The-Huginn** [comment]
I will update this PR soon.

Redoing the logic, to validate ids on startup of the application once.

File: src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java [hunk 2]
```diff
@@ -1,21 +1,55 @@
...
 public class TriagePullRequestProcessor {
     private static final Logger LOG = Logger.getLogger(TriagePullRequestProcessor.class);
 
+    @Inject
+    GitHubClientProvider clientProvider;
+
+    @Inject
+    GitHubConfigFileProvider fileProvider;
+
+    @PostConstruct
```
Comment (xstefank) [inline src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java:34]: do it with `@Observes StartupEvent` as it is recommended.

File: src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java [hunk 3]
```diff
@@ -1,21 +1,55 @@
...
+    void checkConfiguration() {
+        try {
+            for (GHAppInstallation installation : clientProvider.getApplicationClient().getApp().listInstallations()) {
+                GitHub app = clientProvider.getInstallationClient(installation.getId());
+                for (GHRepository repository : app.getInstallation().listRepositories()) {
+                    WildFlyConfigFile wildflyBotConfigFile = fileProvider.fetchConfigFile(repository, "wildfly-bot.yml", ConfigFile.Source.DEFAULT, WildFlyConfigFile.class).get();
```
Comment (xstefank) [inline src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java:40]: We should put the name of the config to some shared constant somewhere so it doesn't need to be typed in string everywhere and prone to errors.

File: src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java [hunk 4]
```diff
@@ -1,21 +1,55 @@
...
+                    for (WildFlyRule rule : wildflyBotConfigFile.wildfly.rules) {
+                        if (rule.id == null) {
+                            LOG.errorf("Following rule is missing an id. [%s]", rule);
```
Comment (xstefank) [inline src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java:43]: Maybe collect the names and put just one message listing all.

File: src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java (hunk 2 above)
Comment (xstefank) [reply src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java:34]: And put it into a separate class. Don't mix two functionalities in one class.

File: src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java [hunk 5]
```diff
@@ -28,12 +62,17 @@ void onPullRequestOpened(@PullRequest.Opened GHEventPayload.PullRequest pullRequ
...
+                if (Matcher.matches(pullRequest, rule)) {
+                    LOG.debugf("Matched rule with id: %s.", rule.id);
+                    for (String nick : rule.notify) {
+                        if (!nick.equals(pullRequest.getUser().getLogin())) {
+                            mentions.add(nick);
+                        }
                     }
                 }
+            } else {
+                LOG.warn("Rule is missing an id.");
```
Comment (xstefank) [inline src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java:75]: So we need to decide if it's an error or a warning. Do we need to log message on every PR processing? I think that on startup is enough. Well, we should also execute the (current) postconstruct method when a new repo registers. Can you add that please?

File: src/main/java/io/xstefank/wildlfy/bot/config/WildFlyConfigFile.java [hunk 6]
```diff
@@ -30,5 +32,13 @@ public static final class WildFlyRule {
 
         @JsonDeserialize(as = TreeSet.class)
         public Set<String> notify = new TreeSet<>();
+
+        private String stringify(String value) {
```
Comment (xstefank) [inline src/main/java/io/xstefank/wildlfy/bot/config/WildFlyConfigFile.java:36]: private methods are after public ones

File: src/test/java/io/xstefank/wildfly/bot/RuleMissingIdTest.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+                "wildfly-bot.yml", """
+                    wildfly:
+                      rules:
+                        - title: "Test"
+                          notify: [7125767235,0979986727]
+                    """))
+        .when().payloadFromClasspath("/pr-opened.json")
+        .event(GHEvent.PULL_REQUEST)
+        .then().github(mocks -> {
+            verifyNoMoreInteractions(mocks.ghObjects());
```
Comment (xstefank) [inline src/test/java/io/xstefank/wildfly/bot/RuleMissingIdTest.java:29]: you might want to try to catch the message log - https://github.com/smallrye/smallrye-health/blob/4c79455b2e78c42a9a745cbabbf7e67c5fde4fd6/implementation/src/test/java/io/smallrye/health/SmallRyeHealthReporterTest.java#L44

**Review by xstefank — CHANGES_REQUESTED** [review]

File: src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java (hunk 3 above)
Comment (The-Huginn) [reply src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java:40]: I have opened an issue for this please refer to #53

File: src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java (hunk 5 above)
Comment (The-Huginn) [reply src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java:75]: Should I open a new issue for that?

**Comment by The-Huginn** [comment]
I have addressed most comments.
However I realized, that the test is started with Github AppId in .env file and the initial checks are fired against that app. I have not digged into mocking our own app if even possible. Also the startup event is seemingly fired before the tests could register custom log handler for capturing logs.

File: src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java (hunk 5 above)
Comment (xstefank) [reply src/main/java/io/xstefank/wildlfy/bot/TriagePullRequestProcessor.java:42]: No, just remove it here to remove duplications

File: src/main/java/io/xstefank/wildlfy/bot/LifecycleProcessor.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+import org.jboss.logging.Logger;
+import org.kohsuke.github.GHAppInstallation;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class LifecycleProcessor {
```
Comment (xstefank) [inline src/main/java/io/xstefank/wildlfy/bot/LifecycleProcessor.java:20]: I'm surprised this works. Add `@ApplicationScoped` here.

File: src/main/java/io/xstefank/wildlfy/bot/config/RuntimeConfiguration.java [hunk 9]
```diff
@@ -0,0 +1,6 @@
+package io.xstefank.wildlfy.bot.config;
+
+public class RuntimeConfiguration {
```
Comment (xstefank) [inline src/main/java/io/xstefank/wildlfy/bot/config/RuntimeConfiguration.java:3]: Constants

File: src/main/java/io/xstefank/wildlfy/bot/config/RuntimeConstants.java [hunk 10]
```diff
@@ -1,6 +1,6 @@
 package io.xstefank.wildlfy.bot.config;
 
-public class RuntimeConfiguration {
+public class RuntimeConstants {
 
     public static final String configFile = "wildfly-bot.yml";
```
Comment (xstefank) [inline src/main/java/io/xstefank/wildlfy/bot/config/RuntimeConstants.java:5]:
```suggestion
    public static final String CONFIG_FILE_NAME = "wildfly-bot.yml";
```

**Review by xstefank — CHANGES_REQUESTED** [review]

**Review by xstefank — APPROVED** [review]

**Comment by The-Huginn** [comment]
Resolved conflicts with main. Squashed into one commit.

Should be ready to merge.

**Comment by The-Huginn** [comment]
Sorry, I didn't restart Quarkus to run wildfly-checkstyle plugin

**Review by xstefank — APPROVED** [review]

**Pull request merged**